dashi/poolqc/pooling_qc_sample.py

```diff
@@ -9,79 +9,159 @@
 import numpy as np
 import urllib
 from datetime import datetime
+import sd_material_ui as sd
 
 rnaseq = pandas.read_hdf('./data/rnaseqqc_cache.hd5')
+# Column name is being renamed to allow for a seamless merge on column 'library'
 rnaseq.rename(columns={'Sample Name': 'library'}, inplace=True)
 
 bamqc = pandas.read_hdf('./data/bamqc_cache.hd5')
 
 bcl2fastq = gsiqcetl.bcl2fastq.parse.load_cache(
     gsiqcetl.bcl2fastq.parse.CACHENAME.SAMPLES,
     './data/bcl2fastq_cache.hd5')
-bcl2fastq['library'] = bcl2fastq['SampleID'].str.extract('SWID_\d+_(\w+_\d+_.*_\d+_[A-Z]{2})_')
+
+# Column is being renamed for clarification
+bcl2fastq.rename(columns={'SampleNumberReads': 'Clusters'}, inplace=True)
+bcl2fastq['library'] = bcl2fastq['SampleID'].str.extract('SWID_\d+_(.+_\d+_[a-zA-Z]{2}_._[A-Z]{2}_\d{3}_[A-Z]{2})_')
```

Comment:
Please replace `.+` with `[A-Z0-9]+`.
Please replace `\d{3}` with `\d+` as it is a variable number of digits.